ResearchSuite/ResearchSuite/RSDMotionRecorder.swift

```diff
@@ -0,0 +1,585 @@
+///
+/// You will need to add the privacy permission for  motion sensors to the application `Info.plist`
+/// file. As of this writing (syoung 02/09/2018), the required key is:
+/// - `Privacy - Motion Usage Description`
+///
+/// - note: This recorder is only available on iOS devices. CoreMotion is not supported by other
+///         platforms.
+///
+/// - seealso: `RSDMotionRecorderType`, `RSDMotionRecorderConfiguration`, and `RSDMotionRecord`.
+@available(iOS 10.0, *)
+public class RSDMotionRecorder : RSDSampleRecorder {
+
+    /// The most recent device motion sample. This property is updated on the motion queue.
+    /// This is an `@objc dynamic` property so that listeners can be set up using KVO to
+    /// observe changes to this property.
+    @objc dynamic public private(set) var currentDeviceMotion: CMDeviceMotion?
+    
+    /// The most recent accelerometer data sample. This property is updated on the motion queue.
+    /// This is an `@objc dynamic` property so that listeners can be set up using KVO to
+    /// observe changes to this property.
+    @objc dynamic public private(set) var currentAccelerometerData: CMAccelerometerData?
+    
+    /// The most recent gyro data sample. This property is updated on the motion queue.
+    /// This is an `@objc dynamic` property so that listeners can be set up using KVO to
+    /// observe changes to this property.
+    @objc dynamic public private(set) var currentGyroData: CMGyroData?
+    
+    /// The most recent magnetometer data sample. This property is updated on the motion queue.
+    /// This is an `@objc dynamic` property so that listeners can be set up using KVO to
+    /// observe changes to this property.
+    @objc dynamic public private(set) var currentMagnetometerData: CMMagnetometerData?
+    
+    /// The motion sensor configuration for this recorder.
+    public var motionConfiguration: RSDMotionRecorderConfiguration? {
+        return self.configuration as? RSDMotionRecorderConfiguration
+    }
+    
+    /// The recorder types to use for this recording. This will be set to the `recorderTypes`
+    /// from the `coreMotionConfiguration`. If that value is `nil`, then the defaults are
+    /// `[.accelerometer, .gyro]` because all other non-compass measurements can be calculated
+    /// from the accelerometer and gyro.
+    lazy public var recorderTypes: Set<RSDMotionRecorderType> = {
+        return self.motionConfiguration?.recorderTypes ?? [.accelerometer, .gyro]
+    }()
+    
+    /// The sampling frequency of the motion sensors. This will be set to the `frequency`
+    /// from the `coreMotionConfiguration`. If that value is `nil`, then the default sampling
+    /// rate is `100` samples per second.
+    lazy public var frequency: Double = {
+        return self.motionConfiguration?.frequency ?? 100
+    }()
+    
+    /// For best results, only use a single motion manager to handle all motion sensor data.
+    private var motionManager: CMMotionManager?
```

Comment:
should this be a class var so we only create one even if there are multiple motion recorders going at once for some reason?